### PR TITLE
Fix: Edge error locations when using `extends`

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- ~~Added/Updated related documentation.~~
- ~~Added/Updated related tests.~~

## Short description of the change(s)

Improvements in a couple of edge cases:

- `target` was pointing to `{` because it did not do the right search
- when a configuration `extends` but none of the files had a required
  property, `extends` got underlined because it was set as the
  `alternatePath`. With the improvements in the node search this is no
  longer needed.

### CODE REVIEW

**Comment by sarvaje** [comment]
@molant can you upload some screenshots?

**Comment by molant** [comment]
Wrong `target` points to `target` value:

image


Missing `target` points to `compilerOptions`:

image

Wrong `target` in `extends`:

image

Missing `target` in `extends` and current file points to `extends`. This is because the default value is `es3` so we need to point to somewhere and right now the behavior is `extends` (and both files are technically incorrect): 

image

If we don't have `compilerOptions` is not a valid tsconfig.json so it doesn't pass.

**Comment by molant** [comment]
I've also added position to the tests to make sure we don't regress.

**Comment by molant** [comment]
> Do we have tests for the extends property and value?

Yes, they were already in `is-valid` and `no-comments` .
I can add a new one for `target` if you want.

**Comment by sarvaje** [comment]
And those tests are testing also the position (for property and value)? If so, no need to add tests for `target` too.

**Comment by molant** [comment]
Yes, they test for the position but I'm adding the `target` ones as well as they use a different logic.

**Comment by sarvaje** [comment]
Ok, perfect!

**Comment by molant** [comment]
Done

File: packages/hint-typescript-config/src/helpers/config-checker.ts [hunk 1]
```diff
@@ -15,6 +15,38 @@ const findValue = (property: string, config: TypeScriptConfig) => {
...
+
+    const ancestors = propertyPath.split('.');
+
+    ancestors.pop();
+
+    if (typeof valueInOriginal !== 'undefined') {
+
+        location = getLocation(propertyPath, { at: 'value' });
+
+    } else if (targetValue !== valueInOriginal &&
```
Comment (antross) [inline packages/hint-typescript-config/src/helpers/config-checker.ts]: Given `valueInOriginal` is `undefined` by this point do we even need to compare?
```suggestion
    } else if (
```

Comment (molant) [reply packages/hint-typescript-config/src/helpers/config-checker.ts]: Done

**Comment by molant** [comment]
@antross do you have the right timestamp or is my machine the broken one?
image

**Comment by antross** [comment]
@molant must be your machine - your commit to fix linting appears even before my review comments from earlier in the day...

image

**Comment by antross** [comment]
I just combined related commits. Will merge once CI passes.

**Comment by antross** [comment]
@sarvaje Did you want to give this one final look? We're hoping to merge before releasing today if possible.
